### PR TITLE
Create @bbc/gel-foundations-styled-components

### LINKED ISSUE #45: GEL-Styled-Components package
**Is your feature request related to a problem? Please describe.**
We should create a `gel-styled-components` npm package. 
It should be published to the BBC npm account (instructions here: https://github.com/bbc/npm) 

It should export variables in these files:
```
breakpoints.js
spacing.js
grid.js
typography.js
```

The `breakpoints` and `spacing` files export constants. Naming convention: `GEL_SPACING`
The `grid` and `typography` files export Styled Components. 

We should be able to import constants like this: 

`import { GEL_SPACING, GEL_MARGIN_BELOW_400PX } from "@bbc/gel-styled-components/spacing"`
`import { GROUP_0_SCREEN_WIDTH_MIN } from "@bbc/gel-styled-components/breakpoints"`
`import { T_BREVIER } from "@bbc/gel-styled-components/typography"`

Full discussion was on this issue: https://github.com/BBC-News/psammead/issues/32

**Describe the solution you'd like**
The package should live in this repository, in this directory:
`packages/utilities/psammead-styles`

It should have this structure:

- `README.md` (Basic description)
- `package.json` (Author: psammeadmaintainers@bbc.co.uk)
- `CONTRIBUTING.md` (for now, it should link to a parent CONTRIBUTING.md in `psammead`

- `spacing.js` Reference from Simorgh repo: https://github.com/BBC-News/simorgh/blob/latest/src/app/lib/constants/styles.js#L30-L50
- `breakpoints.js` Reference from Simorgh repo: https://github.com/BBC-News/simorgh/blob/latest/src/app/lib/constants/styles.js#L51-L82  Note: all these constants should be all-caps with underscores. e.g. `GROUP_0_SCREEN_WIDTH_MIN` 
The three typography breakpoint values should be updated to be in line with the GEL guidelines here http://www.bbc.co.uk/gel/guidelines/typography#type-sizes
`smartPhoneScreenWidthMin` => `GROUP_B_MIN_WIDTH`
`smartPhoneScreenWidthMax` => `GROUP_B_MAX_WIDTH`
`laptopScreenWidthMin` => `GROUP_CD_MIN_WIDTH`

As a result, this `mediaQuery` should be updated to be `MEDIA_QUERY_TYPOGRAPHY`. It is only needed for Typography https://github.com/BBC-News/simorgh/blob/latest/src/app/helpers/mediaQueries.js#L7-L13

- `typography.js`: Reference from Simorgh repo: https://github.com/BBC-News/simorgh/blob/latest/src/app/lib/constants/typography.js#L12-L73

- `grid.js` Reference from Simorgh repo: https://github.com/BBC-News/simorgh/blob/latest/src/app/lib/layoutGrid.js#L16-L82  **I suggest not yet including the grid variables in this PR.** These four variables `layoutWrapperWithoutGrid` `layoutGridWrapper` `layoutGridItemConstrained` `layoutGridItem` are currently only used in Footer, Brand, Article components and they should be refactored before we add them to this utility package. 

**Describe alternatives you've considered**
N/A

**Testing notes**
We should be able to import constants like this: 
`import { T_LONG_PRIMER } from "@bbc/gel-styled-components/typography"`
`import { T_LONG_PRIMER } from "@bbc/gel-styled-components/typography"`

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Builds on work started in https://github.com/BBC-News/psammead/pull/48

Resolves #45

Creates @bbc/gel-foundations-styled-components which would include typography.

This does not include grid constants as they will be pulled in at a later date.

- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by dr3** [comment]
This PR is blocked until https://github.com/BBC-News/psammead/pull/70 is merged as this requires the package to be public

**Comment by bcmn** [comment]
No longer blocked as #70 is merged.

**Comment by dr3** [comment]
This Package wont work until this package and `@bbc/gel-constants` are babelified, however as this is also the case for the other packages, I think this is ok to merge after review

**Review by bcmn — DISMISSED** [review]

File: packages/utilities/gel-foundations-styled-components/package.json
Comment (bcmn) [inline packages/utilities/gel-foundations-styled-components/package.json]: I think our package names & descriptions should be more descriptive than this. _"What is GEL? Sure, this contains styled components, but what of?"_

File: .eslintignore
Comment (bcmn) [inline .eslintignore]: Will the addition of these make the two immediately above redundant?

**Comment by dr3** [comment]
👀 at jenkins failure

File: packages/utilities/gel-foundations-styled-components/README.md
Comment (bcmn) [inline packages/utilities/gel-foundations-styled-components/README.md]: Again, not hugely descriptive. My understanding: this package is intended to contain the GEL grid & typography. On the whole "shareability" front, I think we need to be really, really good about how we name, document & present our packages to people who've never seen them before if we ever want to convince them to use them. 🤔

Comment (dr3) [reply packages/utilities/gel-foundations-styled-components/README.md]: Talked on slack, made a change but happy for @bcmn to suggest a change

**Review by bcmn — DISMISSED** [review]

**Comment by dr3** [comment]
Raised https://github.com/BBC-News/psammead/issues/77 to come back and improve the GEL descriptions as agreed with @bcmn

File: lerna.json
Comment (ChrisBAshton) [inline lerna.json]: This slipped in, I'm guessing? Can be removed from the PR

File: packages/utilities/gel-foundations-styled-components/README.md
Comment (ChrisBAshton) [inline packages/utilities/gel-foundations-styled-components/README.md]:
```suggestion
`/typography` - GEL typography. These are based on the gel typography standard which can be found in the [GEL guidelines](https://www.bbc.co.uk/gel/guidelines/typography).
```

File: packages/utilities/gel-foundations-styled-components/typography.js
Comment (ChrisBAshton) [inline packages/utilities/gel-foundations-styled-components/typography.js]: Didn't we change these constants? `MEDIA_QUERY_TYPOGRAPHY.SMARTPHONE_AND_LARGER`?

**Review by ChrisBAshton — COMMENTED** [review]
Some comments - biggest concern is the `MEDIA_QUERY_TYPOGRAPHY` object - please check!

Otherwise looks great.

File: lerna.json
Comment (dr3) [reply lerna.json]: It needs to be added so the `packages/utilities/gel-foundations-styled-components` is installed, so that eslint doesnt complain with the dependancy see https://ci.news.tools.bbc.co.uk/job/psammead/job/gel-foundations-styled-components/10/display/redirect

File: packages/utilities/gel-foundations-styled-components/typography.js
Comment (dr3) [reply packages/utilities/gel-foundations-styled-components/typography.js]: Fixed

File: packages/utilities/gel-foundations-styled-components/typography.js
Comment (sareh) [inline packages/utilities/gel-foundations-styled-components/typography.js]: This should be `import { MEDIA_QUERY_TYPOGRAPHY } from '@bbc/gel-constants/breakpoints';`, since media query typography isn't the default export of the breakpoints file.

**Review by sareh — COMMENTED** [review]
On an `npm run build`, I get this error `src doesn't exist`
You should create this directory: `/packages/utilities/gel-foundations-styled-components/src`.

I've also noticed that the `src` directory needs to be created in all the other packages too. ~I'll open a separate issue for that.~ I've noticed it is in this PR. https://github.com/BBC-News/psammead/pull/74.

**Comment by dr3** [comment]
Talked to @sareh on slack, We will merge #74 and then update this PR to use the new package pattern for building

**Review by sareh — APPROVED** [review]
👍 Looks good!

**Comment by radiocontrolled** [comment]
LGTM 👍

**Review by pjlee11 — APPROVED** [review]

**Pull request merged**
